frontend/app/work_packages/services/work-package-field-service.js

```diff
@@ -91,8 +91,15 @@ module.exports = function(
     return isEmpty(workPackage, field);
   }
 
+  function isMilestone(workPackage) {
+    return workPackage.form.embedded.payload.links.type.props.href === '/api/v3/types/4';
```

Comment:
WTF? A milestone is a type which has its milestone property set to true.

Thanks for @Azure7111 for pointing me to this (with an already outdated comment). Neither the name, nor the ID are determinators whether the type is a milestone type. There can be more than one btw...